lib/assessment/label.py

```diff
@@ -358,15 +385,26 @@ def compute_messages(self, prompt, rubric, student_code, examples=[]):
         return messages
 
     def get_response_data_if_valid(self, choice, rubric, student_id, choice_index=None, reraise=False, response_type='tsv'):
-        response_text = choice['message']['content']
+        response_text = None
+        finish_reason = None
+        if 'message' in choice.keys(): # OpenAI response
+            response_text = choice['message']['content']
+            finish_reason = choice['finish_reason']
+        elif 'completion' in choice.keys(): # Claude 2 response
+            response_text = choice['completion']
+            finish_reason = choice['stop_reason']
+        elif 'content' in choice.keys(): # Claude 3 response
+            response_text = choice['content'][0]['text']
+            finish_reason = choice['stop_reason']
```

Comment:
Non blocking, but I'm picking up a code smell here that makes me a little nervous about the future maintainability of this code if we need to add/switch models frequently. It seems like we're choosing our model at the beginning of the flow, but then 'forgetting' what model we're using and examining the responses to infer the model and how to parse it's responses.

I wonder if it would be beneficial to add an abstraction layer here to shove model/provider specific code into dedicated classes that accept the same inputs and return normalized outputs. The choice is between having some code duplication across various classes or scattering conditionals throughout the code - hard to say what the best way is right now. If we start doing shenanigans like `if looks_like('model4') || is_kinda('model3') then response.tolowercase` then it might be time to refactor.

Reply:
I agree with this feedback. one idea is to use an abstract base class to represent the model which various specific LLMs inherit from. in any case, let's consider this as potential follow-up work.

Reply:
I think that's a good idea. Most of this code was written specifically for connecting with openai, so it would be good to do a refactor. I'll add it as a follow up jira task that we can possibly tackle after the release this month.